Extended dark markers to include blobs, rather than just a tesselation

richardbeare opened the pull request:
I've added an extra marker step, hopefully making the result appropriate for CnJ. The new step places marker on dark tissue, and is intended to provide a RBC class. Of course it also places markers in other dark regions, thus splitting those regions into more pieces than previously.

I'm still deleting the original dark class from the result, and you may want to retain it, depending on the CnJ strategy.

Hopefully this procedure is more stable and faster than the minima preflooding approach as the constraints are different for light and dark markers, and are mostly spatial.
Enjoy

romangrothausmann committed:
ENH: stop-value for fast-marching in order to avoid INF values in output

richardbeare committed:
Extended dark markers to include blobs, rather than just a tesselation

romangrothausmann committed:
BUG: cast needed for calc with InputComponentType > int16_t (to avoid overflow warning)

romangrothausmann committed:
PERF: avoid maskFiler by using SetBinaryMask()

romangrothausmann committed:
Revert "PERF: avoid maskFiler by using SetBinaryMask()"

This reverts commit 29af7713ad07e084b94d7bf9eec1af836d0ef543

test showed that
  output is exactly the same but
  used more than twices as much memory and
  did not run noticeably faster

romangrothausmann committed:
itkCountNeighborsImageFilter as submodule

romangrothausmann commented:
Many thanks Richard, compiles and runs fine, but I've trouble finding suitable parameters. I ran e.g.
`segmentLung -i s02_bi3_rs+1+150_smROI_ROI001.mha -o  s02_bi3_rs+1+150_smROI_ROI001_RBseg2.mha --markervol 100000 --markererode 15 --debugfolder RBseg2/ -d --gradsigma 0.5 --darkmarkerdilate 10`
but have problems such as e.g. at 1632 285 501 the RBC has the same label (21640) as a large BG region while at 1632 410 545 an RBC has label 0, similar at many other RBCs or other obstacles.

richardbeare commented:
I was using darkmarkerdilate 5 in my test, but I haven't looked as closely at the results as you. I think 10 might completely remove some RBCs, leaving them without their own marker.

This approach depends on some isolation between the RBC and the neighbouring structures, such that the connection can be broken by the dilation. If the RBC is pack tightly against the background structure it will probably remain joined. I presume these structures were joined in the previous attempt as well?

One thing I didn't do this time was remove large dark markers - I suspect if won't correct this problem completely, but is easy to do and might simplify a CnJ procedure.

Finally - I haven't used brightness in any sophisticated way for the RBC attempt. Do you expect a consistent brightness range for RBCs? If so then it might simplify matters.

richardbeare committed:
Size threshold on simple dark marker

richardbeare committed:
regional minima for RBC markers

romangrothausmann committed:
Merge branch 'master' into LungSeg2

romangrothausmann committed:
cout for stat result

romangrothausmann committed:
comment on correspondence of vo2ve.cxx <-> vo-skel2poly.cxx (VTK, vo2ve.git)

richardbeare committed:
Including all dark areas, rather than attempting to automatically
remove redundant ones. Labelling the results of stage 1 markers
so that they can be joined with greater flexibility.

romangrothausmann committed:
Merge branch 'LungSeg2' of https://github.com/richardbeare/ITK-CLIs into LungSeg2

romangrothausmann committed:
taking template.cxx as base for project.cxx

romangrothausmann committed:
basic implementation of project.cxx (mean)

romangrothausmann committed:
taking template.cxx as base for template_var-filter.cxx

romangrothausmann committed:
basic changes to make the filter variable

romangrothausmann committed:
project.cxx based on template_var-filter.cxx

romangrothausmann committed:
project.cxx with all variants of itkProjectionImageFilter (some might need further adaptions)

romangrothausmann committed:
basic implementation of channel_extract, RGB and complex need special treatment

romangrothausmann committed:
taking template_vec.cxx as base for channel_extract.cxx

romangrothausmann committed:
taking RGB2vec.cxx as base for vec2RGB.cxx

romangrothausmann committed:
taking template_vec.cxx as base for vec2RGB.cxx

romangrothausmann committed:
basic implementation of vec2RGB.cxx

romangrothausmann committed:
removed variable length vector pixelType from template.cxx, use template_vec.cxx instead if needed

romangrothausmann committed:
RGB works as is, assuming RGBA as well

romangrothausmann committed:
interpreting COMPLEX pixelType as vector of 2 components (not tested)

romangrothausmann committed:
moved COMPLEX (2 components) before RGB (3 components)

romangrothausmann committed:
merge new template_2inputs.cxx into add.cxx

romangrothausmann committed:
taking add.cxx as base for subtract.cxx

romangrothausmann committed:
basic implementation of subtract.cxx

romangrothausmann committed:
InputComponent1 should be InputComponentType1 (was not used so far)

romangrothausmann committed:
subtract now able to switch to signed types even for unsigned inputs

romangrothausmann committed:
signed types need casting

romangrothausmann committed:
Merge branch 'master' into LungSeg2
